src/test/java/com/google/sps/servlets/TestUserServlet.java

```diff
@@ -0,0 +1,123 @@
+    assertTrue(parsedEvent.getTitle().equals("title"));
+    assertTrue(parsedEvent.getOwnerID().equals("123"));
+    assertTrue(parsedEvent.getID().equals("firstId"));
+  }
+
+  @Test
+  public void testUserServlet_doGet_WithInvitedAndJoinedEvents_RemovesJoinedFromInvited()
+      throws IOException, ServletException {
+    fakeUser = new com.google.sps.data.User("123", "email", "username",
+        Arrays.asList(new String[] { "firstId", "secondId" }), Arrays.asList(new String[] { "secondId" }), null);
+
+    when(mockUserService.isUserLoggedIn()).thenReturn(true);
+    when(mockUserService.getCurrentUser()).thenReturn(new User("email", "authDomain", "123"));
+    when(mockUserStorage.getUser("123")).thenReturn(fakeUser);
+    when(mockEventStorage.getEvent("firstId"))
+        .thenReturn(Event.newBuilder().setID("firstId").setOwnerID("123").setTitle("title").build());
+    when(mockEventStorage.getEvent("secondId"))
+        .thenReturn(Event.newBuilder().setID("secondId").setOwnerID("124").setTitle("joinedTitle").build());
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter printWriter = new PrintWriter(stringWriter);
+    when(mockResponse.getWriter()).thenReturn(printWriter);
+
+    new UserServlet(mockUserStorage, mockEventStorage, mockUserService).doGet(mockRequest, mockResponse);
+
+    JsonElement json = JsonParser.parseString(stringWriter.toString());
+    JsonObject jsonObject = json.getAsJsonObject();
+    Gson gson = new Gson();
+    List<LinkedTreeMap> invitedEvents = gson.fromJson(jsonObject.get("invitedEvents"), List.class);
+    List<LinkedTreeMap> joinedEvents = gson.fromJson(jsonObject.get("joinedEvents"), List.class);
+    List<LinkedTreeMap> declinedEvents = gson.fromJson(jsonObject.get("declinedEvents"), List.class);
+    assertTrue(invitedEvents.size() == 1);
+    assertTrue(joinedEvents.size() == 1);
+    assertTrue(declinedEvents.size() == 0);
+    Event parsedJoinedEvent = gson.fromJson(JsonParser.parseString(joinedEvents.get(0).toString()), Event.class);
+    assertTrue(parsedJoinedEvent.getID().equals("secondId"));
+  }
+
+  @Test
+  public void testUserServlet_doPost_WithUserLoggedIn_SetsUsername() throws IOException, ServletException {
+    fakeUser = new com.google.sps.data.User("123", "email", "", Arrays.asList(new String[] { "firstId", "secondId" }),
+        Arrays.asList(new String[] { "secondId" }), null);
+    when(mockUserService.isUserLoggedIn()).thenReturn(true);
+    when(mockRequest.getParameter("nickname")).thenReturn("andru47");
+    when(mockUserService.getCurrentUser()).thenReturn(new User("email", "authDomain", "123"));
+    when(mockUserStorage.getUser("123")).thenReturn(fakeUser);
+    when(mockRequest.getPathInfo()).thenReturn("/username");
+
+    new UserServlet(mockUserStorage, mockEventStorage, mockUserService).doPost(mockRequest, mockResponse);
+
+    assertTrue(fakeUser.getUsername().equals("andru47"));
```

Comment:
You should check whether `userStorageObject.addOrUpdateUser` with the right argument is called.